service/arm_templates/main.json

```diff
@@ -131,9 +180,54 @@
           "masterSubnetCidr": {"value": "[parameters('masterSubnetCidr')]"},
           "workerSubnetCidr": {"value": "[parameters('workerSubnetCidr')]"},
           "masterSecurityGroupID": {"value": "[reference('security_groups_setup').outputs.masterSecurityGroupID.value]"},
-          "workerSecurityGroupID": {"value": "[reference('security_groups_setup').outputs.workerSecurityGroupID.value]"},
+          "workerSecurityGroupID": {"value": "[reference('security_groups_setup').outputs.workerSecurityGroupID.value]"}
         }
       }
     },
+    {
+      "apiVersion": "2016-09-01",
+      "name": "availability_sets_setup",
+      "type": "Microsoft.Resources/deployments",
+      "properties": {
+        "mode": "incremental",
+        "templateLink": {
+          "uri": "[variables('availabilitySetsTemplateURI')]",
```

Comment:
I think we can make availabilitySetsTemplate generic. Like we have in nodesSetupTemplate. We would create one AS for masters and one AS work workers here. Does that makes sense?

Reply:
Note to self: this is done. Commend didn't disappear.